canvases/dark-packages/main.dark

```diff
@@ -0,0 +1,35 @@
+type PackageFns = PACKAGE.Darklang.Stdlib.ProgramTypes.PackageFn.T
+
+[<DB>]
+type PackageFnsDB = PackageFns
+
+[<HttpHandler("POST", "/functions")>]
+let _handler _req =
+  let fn =
+    request.body
+    |> String.fromBytes
+    |> Json.parse<PACKAGE.Darklang.Stdlib.ProgramTypes.PackageFn.T>
+    |> unwrap
+
+  let fnToSave =
+    PackageFns
+      { tlid = fn.tlid
+        id = fn.id
+        name = fn.name
+        body = fn.body
+        typeParams = fn.typeParams
+        parameters = fn.parameters
+        returnType = fn.returnType
+        description = fn.description
+        deprecated = fn.deprecated }
+
+  let key = DB.generateKey_v0 () //use id instead
+  DB.set fnToSave key PackageFnsDB
+
+  PACKAGE.Darklang.Stdlib.Http.response (String.toBytes "added") 200
+
+[<HttpHandler("GET", "/all-functions")>]
+let _handler _req =
+  let count = (DB.count PackageFnsDB) |> Int.toString
```

Comment:
I tried the following to see what we are saving, but I encountered an `unknown error`. The `DB.count` function returns 107, indicating that we are saving something. However, I'm unsure if we are saving the right things. (The same applies to DB.getAllWithKeys)

```
[<HttpHandler("GET", "/functions")>]
let _handler _req =
   let functions = (DB.getAll PackageFnsDB) |> String.fromList 

  PACKAGE.Darklang.Stdlib.Http.response (String.toBytes functions) 200
  ```

Reply:
It would be good to investigate/resolve this as a followup.